lib/lotus/model/configuration.rb

```diff
@@ -137,6 +144,101 @@ def mapping(path=nil, &blk)
         @mapper_config = Lotus::Model::Config::Mapper.new(path, &blk)
       end
 
+      # Set the logger
+      #
+      # @param logger [Logger] the logger instance
+      #
+      # @example Set custom logger and retrieve the set logger
+      #   require 'lotus/model'
+      #
+      #   Lotus::Model.configure do
+      #     logging ::Logger.new(STDOUT)
+      #   end
+      #
+      #   Lotus::Model.configuration.logging
+      #     # => #<Logger:0x007ffc45227920>
+      #
+      #   Lotus::Model.configuration.logger
+      #     # => #<Logger:0x007ffc45227920>
+      #
+      # @since 0.3.0
+      # @see Lotus::Model.configure
+      def logging(logger = nil)
+        if logger.nil?
+          @logger
+        else
+          @logger ||= logger
+        end
+      end
+
+      # Set directory where migration files should be migrated from
+      #
+      # @param directory [String] the migration directory filepath
+      #
+      # @example Set custom logger and retrieve the set logger
+      #   require 'lotus/model'
+      #
+      #   Lotus::Model.configuration.migration_directory
+      #     # => 'db/migrations'
+      #
+      #   Lotus::Model.configure do
+      #     migration_directory 'my_custom/migrations'
+      #   end
+      #
+      #   Lotus::Model.configuration.migration_directory
+      #     # => 'my_custom/migrations'
+      #
+      # @since 0.3.0
+      # @see Lotus::Model.configure
+      def migration_directory(directory=nil)
+        if directory.nil?
+          @migration_directory ||= _default_migration_directory
+        else
+          @migration_directory = directory
+        end
+      end
```

Comment:
@plukevdh I believe this is a convention that @jodosha has established. Namely, for configuration options, the named configuration option (i.e. `migration_directory`) should be both setter and getter. Consider the following:

```ruby
Lotus::Model.configure do
  migration_directory 'my_custom/migrations'
end
```

If you have a setter, then I believe you will need an explicit receiver, otherwise you are creating a local variable (and not assigning the value to the configuration).

It is a very valid question, and I don't know if @jodosha can point towards a "convention" document. Or if that is something part of the CONTRIBUTING document.